addons/website_sale_collect/static/src/js/variant_mixin.js

```diff
@@ -8,7 +8,14 @@ const oldOnChangeCombinationStock = VariantMixin._onChangeCombinationStock;
  * @override of `website_sale_stock`
  */
 VariantMixin._onChangeCombinationStock = function (ev, $parent, combination) {
+    const cac_widget = document.querySelector(
+        'owl-component[name="website_sale_collect.ClickAndCollectAvailability"]'
+    );
     if (!combination.show_click_and_collect_availability) {
-        return oldOnChangeCombinationStock.apply(this, arguments);
+        cac_widget?.classList.add('d-none');
     }
+    else {
+        cac_widget?.classList.remove('d-none');
+    }
```

Comment:
Please avoid double negation
```suggestion
    if (combination.show_click_and_collect_availability) {
        cac_widget?.classList.remove('d-none');
    }
    else {
        cac_widget?.classList.add('d-none');
    }
```